uniref-rest/src/main/java/org/uniprot/api/uniref/repository/store/UniRefEntryFacetConfig.java

```diff
@@ -0,0 +1,158 @@
+            this.facetName = facetName;
+            this.label = label;
+        }
+
+        public String getLabel() {
+            return label;
+        }
+
+        public String getFacetName() {
+            return facetName;
+        }
+    }
+
+    @Override
+    public Collection<String> getFacetNames() {
+        return Arrays.stream(UniRefEntryFacet.values())
+                .map(UniRefEntryFacet::getFacetName)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public Map<String, FacetProperty> getFacetPropertyMap() {
+        return null;
+    }
+
+    static List<String> applyFacetFilters(List<String> members, String filter) {
+        List<String> result = new ArrayList<>(members);
+        if (Utils.notNullNotEmpty(filter)) {
+            if (hasFieldTerms(filter, UniRefEntryFacet.MEMBER_ID_TYPE.getFacetName())) {
+                String filterValue =
+                        getTermValue(filter, UniRefEntryFacet.MEMBER_ID_TYPE.getFacetName());
+                result.removeIf(memberValue -> filterMemberTypeValue(filterValue, memberValue));
+            }
+            if (hasFieldTerms(filter, UniRefEntryFacet.UNIPROT_MEMBER_ID_TYPE.getFacetName())) {
+                String filterValue =
+                        getTermValue(
+                                filter, UniRefEntryFacet.UNIPROT_MEMBER_ID_TYPE.getFacetName());
+                result.removeIf(
+                        memberValue -> filterUniProtMemberTypeValue(filterValue, memberValue));
+            }
+        }
+        return result.stream().map(memberId -> memberId.split(",")[0]).collect(Collectors.toList());
+    }
+
+    static List<Facet> getFacets(UniRefEntryLight uniRefEntryLight) {
+        List<Facet> results = new ArrayList<>();
+        Facet memberType =
+                Facet.builder()
+                        .label(UniRefEntryFacet.MEMBER_ID_TYPE.getLabel())
+                        .name(UniRefEntryFacet.MEMBER_ID_TYPE.getFacetName())
+                        .allowMultipleSelection(false)
+                        .values(getMemberTypeValues(uniRefEntryLight.getMembers()))
+                        .build();
+        if (Utils.notNullNotEmpty(memberType.getValues())) {
+            results.add(memberType);
+        }
+
+        Facet uniProtMemberType =
+                Facet.builder()
+                        .label(UniRefEntryFacet.UNIPROT_MEMBER_ID_TYPE.getLabel())
+                        .name(UniRefEntryFacet.UNIPROT_MEMBER_ID_TYPE.getFacetName())
+                        .allowMultipleSelection(false)
+                        .values(getUniProtMemberTypeValues(uniRefEntryLight.getMembers()))
+                        .build();
+        if (Utils.notNullNotEmpty(uniProtMemberType.getValues())) {
+            results.add(uniProtMemberType);
+        }
+
+        return results;
+    }
+
+    private static boolean filterMemberTypeValue(String filterValue, String memberValue) {
```

Comment:
Please put some comment here as well since it is negation of what one can think of.